arbnode/transaction_streamer.go

```diff
@@ -198,6 +198,37 @@ func deleteStartingAt(db ethdb.Database, batch ethdb.Batch, prefix []byte, minKe
 	return iter.Error()
 }
 
+// deleteFromRange deletes key ranging from startMinKey(inclusive) to endMinKey(exclusive)
+func deleteFromRange(db ethdb.Database, prefix []byte, startMinKey uint64, endMinKey uint64) ([][]byte, error) {
+	batch := db.NewBatch()
+	startIter := db.NewIterator(prefix, uint64ToKey(startMinKey))
+	endKey := dbKey(prefix, endMinKey)
+	defer startIter.Release()
+	var prunedKeys [][]byte
+	for startIter.Next() {
+		if bytes.Equal(startIter.Key(), endKey) {
```

Comment:
If I understand this correctly, when endMinKey is missing it will keep deleting values after it.
We should instead extract the index and stop if it is >= endMinKey

Reply:
and we should add a test that when endKey is missing we don't delete anything above it.

Reply:
done.